runtime/commonMain/src/kotlinx/serialization/internal/NullableSerializer.kt

```diff
@@ -5,50 +5,64 @@
 package kotlinx.serialization.internal
 
 import kotlinx.serialization.*
-import kotlin.reflect.KClass
 
-fun <T : Any> makeNullable(actualSerializer: KSerializer<T>): KSerializer<T?> = NullableSerializer(actualSerializer)
 
-class NullableSerializer<T : Any>(public val actualSerializer: KSerializer<T>) : KSerializer<T?> {
-    private class SerialDescriptorForNullable(val original: SerialDescriptor): SerialDescriptor by original {
-        override val isNullable: Boolean
-            get() = true
-
-        override fun equals(other: Any?): Boolean {
-            if (this === other) return true
-            if (other !is SerialDescriptorForNullable) return false
-
-            if (original != other.original) return false
+/**
+ * Returns a nullable serializer for the given serializer of non-null type.
+ */
+public fun <T : Any> KSerializer<T>.nullable(): KSerializer<T?> {
+    return NullableSerializer(this)
+}
 
-            return true
-        }
+@Deprecated(
+    message = "Deprecated in the favor of extension",
+    level = DeprecationLevel.WARNING,
+    replaceWith = ReplaceWith("actualSerializer.nullable()")
+)
+fun <T : Any> makeNullable(actualSerializer: KSerializer<T>): KSerializer<T?> {
```

Comment:
Note that this declaration is used directly by JVM compiler plugin, so it is impossible to remove it for now without plugin change.

Reply:
Sure. Could you please take care of it?

Reply:
OK, just leave it deprecated, I'll remove it later